Modules/vc-module-imageTools/VirtoCommerce.ImageToolsModule.Data/Services/ThumbnailTaskSearchService.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using VirtoCommerce.ImageToolsModule.Core.Models;
+using VirtoCommerce.ImageToolsModule.Core.Services;
+using VirtoCommerce.ImageToolsModule.Data.Models;
+using VirtoCommerce.ImageToolsModule.Data.Repositories;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.ImageToolsModule.Data.Services
+{
+    public class ThumbnailTaskSearchService : IThumbnailTaskSearchService
+    {
+        private readonly Func<IThumbnailRepository> _thumbnailRepositoryFactory;
+
+        public ThumbnailTaskSearchService(Func<IThumbnailRepository> thumbnailThumbnailRepositoryFactoryFactory)
+        {
+            _thumbnailRepositoryFactory = thumbnailThumbnailRepositoryFactoryFactory;
+        }
+
+        public async Task<GenericSearchResult<ThumbnailTask>> SearchAsync(ThumbnailTaskSearchCriteria criteria)
+        {
+            using (var repository = _thumbnailRepositoryFactory())
+            {
+                var query = GetTasksQuery(repository, criteria);
+
+                var sortInfos = criteria.SortInfos;
+                if (sortInfos.IsNullOrEmpty())
+                {
+                    sortInfos = new[]
+                    {
+                        new SortInfo
+                        {
+                            SortColumn = ReflectionUtility.GetPropertyName<ThumbnailTask>(t => t.CreatedDate), SortDirection = SortDirection.Descending
+                        }
+                    };
+                }
+                query = query.OrderBySortInfos(sortInfos);
+                var totalCount = await query.CountAsync();
+
+                var ids = await query.Skip(criteria.Skip).Take(criteria.Take).Select(x => x.Id).ToArrayAsync();
+                var thumbnailTasks = await repository.GetThumbnailTasksByIdsAsync(ids);
+                var results = thumbnailTasks.Select(t => t.ToModel(AbstractTypeFactory<ThumbnailTask>.TryCreateInstance())).ToArray();
```

Comment:
ToArrayAsync

Reply:
The method uses the implementation ToArrayAsync. This I uses linq